src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs

```diff
@@ -789,145 +789,146 @@ public static object EvtGetLogInfo(EventLogHandle handle, UnsafeNativeMethods.Ev
 
         public static void EvtRenderBufferWithContextSystem(EventLogHandle contextHandle, EventLogHandle eventHandle, UnsafeNativeMethods.EvtRenderFlags flag, SystemProperties systemProperties)
         {
-            IntPtr buffer = IntPtr.Zero;
-            IntPtr pointer = IntPtr.Zero;
-            int bufferNeeded;
-            int propCount;
-
-            try
+            unsafe
             {
-                bool status = UnsafeNativeMethods.EvtRender(contextHandle, eventHandle, flag, 0, IntPtr.Zero, out bufferNeeded, out propCount);
-                if (!status)
+                IntPtr buffer = IntPtr.Zero;
+                int bufferNeeded;
+                int propCount;
+
+                try
                 {
-                    int error = Marshal.GetLastWin32Error();
-                    if (error != Interop.Errors.ERROR_INSUFFICIENT_BUFFER)
-                        EventLogException.Throw(error);
-                }
+                    bool status = UnsafeNativeMethods.EvtRender(contextHandle, eventHandle, flag, 0, IntPtr.Zero, out bufferNeeded, out propCount);
+                    if (!status)
+                    {
+                        int error = Marshal.GetLastWin32Error();
+                        if (error != Interop.Errors.ERROR_INSUFFICIENT_BUFFER)
+                            EventLogException.Throw(error);
+                    }
 
-                buffer = Marshal.AllocHGlobal((int)bufferNeeded);
-                status = UnsafeNativeMethods.EvtRender(contextHandle, eventHandle, flag, bufferNeeded, buffer, out bufferNeeded, out propCount);
-                int win32Error = Marshal.GetLastWin32Error();
-                if (!status)
-                    EventLogException.Throw(win32Error);
+                    buffer = Marshal.AllocHGlobal((int)bufferNeeded);
+                    status = UnsafeNativeMethods.EvtRender(contextHandle, eventHandle, flag, bufferNeeded, buffer, out bufferNeeded, out propCount);
+                    int win32Error = Marshal.GetLastWin32Error();
+                    if (!status)
+                        EventLogException.Throw(win32Error);
 
-                pointer = buffer;
-                // Read each Variant structure
-                for (int i = 0; i < propCount; i++)
-                {
-                    UnsafeNativeMethods.EvtVariant varVal = Marshal.PtrToStructure<UnsafeNativeMethods.EvtVariant>(pointer);
```

Comment:
FWIW, getting rid of `PtrToStructure` calls is the main improvement in this change.